pandas/core/generic.py

```diff
@@ -9076,9 +9078,20 @@ def first(self, offset) -> Self:
 
         if len(self.index) == 0:
             return self.copy(deep=False)
+        if type(offset) is not offsets.DateOffset:
```

Comment:
nit: PEP8 suggests using `isinstance` instead of comparing `type` https://peps.python.org/pep-0008/#programming-recommendations

Reply:
Thanks

Reply:
We can't use `isinstance` I think, because:

```
(Pdb) offset
<2 * MonthEnds>
(Pdb) type(offset) is offsets.DateOffset
False
(Pdb) isinstance(offset, offsets.DateOffset)
True
```

Reply:
ah I see, thanks for explaining